src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,368 @@
+
+        task = self._run_task(ahs_program)
+
+        self.samples = task.result()
+
+    def _run_task(self, ahs_program):
+        raise NotImplementedError("Running a task not implemented for the base class")
+
+    def create_ahs_program(self, evolution):
+        """Create AHS program for upload to hardware from a ParametrizedEvolution
+
+        Returns:
+            AnalogHamiltonianSimulation: a program containing the register and drive
+                information for running an AHS task on simulation or hardware"""
+
+        # sets self.pulses to be the evaluated pulses (now only a function of time)
+        self._evaluate_pulses(evolution)
+        self._create_register(evolution.H.register)
+
+        time_interval = evolution.t
+
+        # no gurarentee that global drive is index 0 once we start allowing more just global drive
+        drive = self._convert_pulse_to_driving_field(self.pulses[0], time_interval)
+
+        ahs_program = AnalogHamiltonianSimulation(register=self.register, hamiltonian=drive)
+
+        self.ahs_program = ahs_program
+
+        return ahs_program
+
+    def generate_samples(self):
+        r"""Returns the computational basis samples measured for all wires.
+
+        Returns:
+             array[complex]: array of samples in the shape ``(dev.shots, dev.num_wires)``
+        """
+        return [self._result_to_sample_output(res) for res in self.samples.measurements]
+
+    def _validate_operations(self, operations):
+        """Confirms that the list of operations provided contains a single ParametrizedEvolution
+        from a RydbergHamiltonian with only a single, global pulse"""
+
+        if len(operations) > 1:
+            raise NotImplementedError(
+                f"Support for multiple ParametrizedEvolution operators in a single circuit is "
+                f"not yet implemented. Recieved {len(operations)} operators.")
+
+        ev_op = operations[0]  # only one!
+
+        if not isinstance(ev_op.H, RydbergHamiltonian):
+            raise RuntimeError(
+                f"Expected a RydbergHamiltonian instance for interfacing with the device, but "
+                f"recieved {type(ev_op.H)}.")
+
+        if not set(ev_op.wires) == set(self.wires):
+            raise RuntimeError(f'Device contains wires {self.wires}, but received a `ParametrizedEvolution` operator '
+                               f'working on wires {ev_op.wires}. Device wires must match wires of the evolution.')
+
+        if len(ev_op.H.register) != len(self.wires):
+            raise RuntimeError(f'The defined interaction term has register {ev_op.H.register} of length '
+                               f'{len(ev_op.H.register)}, which does not match the number of wires on the device '
+                               f'({len(self.wires)})')
+
+        self._validate_pulses(ev_op.H.pulses)
+
+    def _validate_pulses(self, pulses):
+        raise NotImplementedError("Validation of pulses not implemented in the base class")
```

Comment:
Another thing to make an `abc.abstractmethod`.

Reply:
- Moved this the base class for now, as we aren't going to try to include local drive options (which would be the difference between hardware and simulator) in the initial PR. Will make it an abstract method when this becomes different for the two devices and moves to the child classes
- Added a docstring specifying that this function checks the list of RydbergPulses and confirms that it describes single, global pulse, and otherwise raises an error